New classes

anthoulab opened the pull request:
Create the classes Employee, Order and PrepArea

anthoulab committed:
New classes

Create the classes Employee, Order and PrepArea

chrisk44 commented on Server/Employee.java:
```diff
@@ -0,0 +1,34 @@
+	protected String password;
+	protected String full_name;
+	protected Date birthdate;
+	protected int amka;
```
ΑΜΚΑ, ΑΦΜ κλπ δεν ειχαμε πει να μη τα βαλουμε?

chrisk44 commented on Server/Order.java:
```diff
@@ -0,0 +1,86 @@
+		waiters.add(w);
+	}
+
+	boolean onEdit(p, action){
```
Για onEdit και send δες τα sequence, ειχαν αλλαξει.

chrisk44 commented on Server/Order.java:
```diff
@@ -0,0 +1,86 @@
+	}
+
+	void setReady(){
+		Waiter.findBestForTable(table);
```
Πρεπει να δημιουργηθει ενα Notification n και μετα:

`Waiter.findBestForTable(table).notify(n);`

μεχρι να επιστρεψει true.

anthoulab committed:
Update class Employee

Delete some unnecessary variables and add the isNew method.

anthoulab committed:
Update Order.java

Update onEdit(), send(), assignOrder(), setReady(), onRejected(), addProduct(). Delete addwaiter(). Add setPaid().

anthoulab committed:
Update PrepArea.java

Update findBestForTable() and getEmployee(). Add showNewOrder().

anthoulab committed:
Update Order.java

Update setReady().

chrisk44 commented on Server/Order.java:
```diff
@@ -0,0 +1,94 @@
+		this.assigned_prepArea = prepArea;
+	}
+
+	void onEdit(p, action){
```
Η onEdit να γίνει onEdit(ArrayList<Product> p, ArrayList<Boolean> action) και γενικά να υλοποιηθεί με βάση το sequence 1.

chrisk44 commented on Server/Order.java:
```diff
@@ -0,0 +1,94 @@
+		}
+	}
+
+	void send(Product p, int action){
```
H send δε παίρνει παραμέτρους και φαίνεται η υλοποίησή της στα sequence 2 και 8.

chrisk44 commented on Server/Order.java:
```diff
@@ -0,0 +1,94 @@
+
+	void assignOrder(PrepArea prepArea, Order o){
+		assigned_prepArea = prepArea;
+		PrepArea.orders.add(o);
```
prepArea αντί για PrepArea, assigned_orders αντί για orders.

chrisk44 commented on Server/PrepArea.java:
```diff
@@ -0,0 +1,30 @@
+	protected LatLng lat_lng;
+	protected Map<Product, int> product_stock;
+	protected int floor;
+	protected ArrayList<Order> orders = new ArrayList();
```
+ ενα assigned_orders

chrisk44 commented on Server/Order.java:
```diff
@@ -0,0 +1,94 @@
+	}
+
+	void setReady(){
+		Notification n;
```
Notification??

chrisk44 commented on Server/Order.java:
```diff
@@ -0,0 +1,94 @@
+
+	}
+
+	void addProduct(){
```
Parameter p

chrisk44 commented on Server/PrepArea.java:
```diff
@@ -0,0 +1,30 @@
+	}
+
+	Employee getEmployee(){
+
```
return

anthoulab committed:
Update Employee.java

anthoulab committed:
Update PrepArea.java

Update getEmployee(). Add List assigned_orders<Order>.

anthoulab committed:
Update PrepArea.java

Change showNewOrder().

anthoulab committed:
Update Order.java

Change onEdit(), assignOrder(), setReady(), addProduct(). Add getBalance().

chrisk44 commented on Server/Order.java:
```diff
@@ -0,0 +1,93 @@
+	void setReady(){
+		Waiter w = Waiter.findBestForTable(this.table);
+		OrderReadyNotification n = new OrderReadyNotification(this, this.assigned_prepArea, w);
+		notify(n);
```
w.notify(n)

chrisk44 commented on Server/Order.java:
```diff
@@ -0,0 +1,93 @@
+	}
+
+	void send(){
+		
```
H υλοποίηση της send() φαίνεται στα sequence 2 και 8.

anthoulab committed:
Update Order.java

anthoulab committed:
Update Order.java

anthoulab committed:
Update PrepArea.java
